src/main/java/no/unit/nva/doi/transformer/DataciteResponseConverter.java

```diff
@@ -93,10 +123,122 @@ private String createDescription() {
+    }
+
+    private PublicationContext extractPublicationContext(DataciteResponse dataciteResponse) throws
+            InvalidIssnException {
+        PublicationType type = getPublicationType(dataciteResponse);
+        if (nonNull(type) && type.equals(JOURNAL_CONTENT)) {
+
+            return new Journal.Builder()
+                    .withPrintIssn(extractPrintIssn(dataciteResponse))
+                    .withTitle(dataciteResponse.getContainer().getTitle())
+                    .withOnlineIssn(extractOnlineIssn(dataciteResponse))
+                    .withPeerReviewed(true)
+                    .withOpenAccess(extractOpenAccess(dataciteResponse))
+                    .withLevel(Level.NO_LEVEL)
+                    .build();
+        }
+        return null;
+    }
+
+    private String extractOnlineIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = getPartOfRelations(dataciteResponse)
+                .stream()
+                .filter(this::isOnlineIssn)
+                .collect(SingletonCollector.collectOrElse(null));
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+
+    }
+
+    private boolean isOnlineIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+                .equals(DataciteRelatedIdentifierType.EISSN);
+    }
+
+    private String extractPrintIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = getPartOfRelations(dataciteResponse)
+                .stream()
+                .filter(this::isPrintIssn)
+                .collect(SingletonCollector.collectOrElse(null));
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+    }
+
+    private List<DataciteRelatedIdentifier> getPartOfRelations(DataciteResponse dataciteResponse) {
+        return dataciteResponse.getRelatedIdentifiers()
+                .stream()
+                .filter(this::isPartOf)
+                .collect(Collectors.toList());
+    }
+
+    private boolean isPrintIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+                .equals(DataciteRelatedIdentifierType.ISSN);
+    }
+
+    private boolean isPartOf(DataciteRelatedIdentifier identifer) {
+        return DataciteRelationType.getByRelation(identifer.getRelationType())
+                .equals(DataciteRelationType.IS_PART_OF);
+    }
+
+    private boolean extractOpenAccess(DataciteResponse dataciteResponse) {
+        List<DataciteRights> rights = Optional.ofNullable(dataciteResponse.getRightsList()).orElse(null);
+        if (nonNull(rights) && !rights.isEmpty()) {
+            return rights.stream().anyMatch(this::hasOpenAccessRights);
+        }
+        return false;
+    }
+
+    private boolean hasOpenAccessRights(DataciteRights dataciteRights) {
```

Comment:
You do not need the AtomicBoolean here. The name `rightsUrlCorrespondsToOpenAccessRightsOrg` is just an example. I do not know what we are looking for exactly here. Better name may be applicable.
```suggestion
  private boolean hasOpenAccessRights(DataciteRights dataciteRights) {
        return Optional.ofNullable(dataciteRights.getRightsUri())
            .map(this::rightsUrlCorrespondsToOpenAccessiIghtsOrg
        ).orElse(Boolean.FALSE);
    }

    private boolean rightsUrlCorrespondsToOpenAccessRightsOrg(String r) {
        return r.contains(CREATIVECOMMONS) || r.equals(INFO_EU_REPO_SEMANTICS_OPEN_ACCESS);
    }
```